pennylane/devices/qubit/measure.py

```diff
@@ -86,17 +87,20 @@ def csr_dot_products(
         TensorLike: the result of the measurement
     """
     total_wires = len(state.shape) - is_state_batched
-    Hmat = measurementprocess.obs.sparse_matrix(wire_order=list(range(total_wires)))
-
-    if is_state_batched:
```

Comment:
I think we need to preserve `if`-section, just after the first `is_pauli_sentence` condition, otherwise non-pauli-sentence observables won't have support for batched states. perhaps a gap in our testing if it didn't pick anything up...

Reply:
Good catch, I'll add a breaking test for that.